packages/protocols/generation-queue.go

```diff
@@ -0,0 +1,127 @@
+package protocols
+
+import (
+	"errors"
+	"time"
+
+	"github.com/GenesisKernel/go-genesis/packages/model"
+
+	"github.com/GenesisKernel/go-genesis/packages/conf/syspar"
+	log "github.com/sirupsen/logrus"
+)
+
+// QueueChecker allow check queue to generate current block
+type QueueChecker interface {
+	TimeToGenerate(position int64) (bool, error)
+	NextTime(position int64, t time.Time) (time.Time, error)
+	BlockForTimeExists(t time.Time, nodePosition int) (bool, error)
+	RangesByTime(t time.Time) (start, end time.Time)
```

Comment:
RangeSByTime, but i see only one range

Reply:
And second, it used only internally in that object, why do we need it in interface?